cli/azd/cmd/auth_login.go

```diff
@@ -368,3 +392,30 @@ func (la *loginAction) login(ctx context.Context) error {
 
 	return nil
 }
+
+func parseUseDeviceCode(ctx context.Context, flag stringPtr, commandRunner exec.CommandRunner) (bool, error) {
+	var useDevCode bool
+
+	explicitDeviceCodeInput := flag.ptr != nil
+	if explicitDeviceCodeInput {
+		userInput, err := strconv.ParseBool(*flag.ptr)
```

Comment:
stylistic nit, YMMV. I would rename `flag` to `useDevCodeFlag`, and the code here would read:

```suggestion
	if useDevCodeFlag != nil {
		userInput, err := strconv.ParseBool(*useDevCodeFlag.ptr)
```